### PR TITLE
Implement ThingIFAPI#listTriggeredServerCodeResult

### PR DESCRIPTION
**Pull request by yongpingchen**
- implement ThingIFAPI#listTriggeredServerCodeResult
- add small tests

### CODE REVIEW

File: thingif/src/test/java/com/kii/thingif/thingifapi/ListTriggeredServerCodeResultTest.java [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+        String triggerID = UUID.randomUUID().toString();
+        TypedID thingID = new TypedID(TypedID.Types.THING, "th.1234567890");
+        String accessToken = "thing-access-token-1234";
+        Target target = new StandaloneThing(thingID.getID(), "vendor-thing-id", accessToken);
+        String paginationKey = "pagination-12345-key";
+
+        ThingIFAPI api = this.createDefaultThingIFAPI(this.context, APP_ID, APP_KEY);
+        ThingIFAPIUtils.setTarget(api, target);
+
+        TriggeredServerCodeResult serverCodeResult1 =
```
Comment (shinsuke-nara) [inline thingif/src/test/java/com/kii/thingif/thingifapi/ListTriggeredServerCodeResultTest.java]: This is cosmetic so you do not need to adopt my suggestion:)

I think following code is more simple.

```java
TriggeredServerCodeResult[] firstResults[] = new TriggeredServerCodeResult[] {
    TriggeredServerCodeResultFactory.create(true, "1234", System.currentTimeMillis(), "func1", null),
    TriggeredServerCodeResultFactory.create(true, "12.34", System.currentTimeMillis() 1000, "func2", null),
    TriggeredServerCodeResultFactory.create(true, "true", System.currentTimeMillis() 2000, "func3", null),
    TriggeredServerCodeResultFactory.create(true, "\"abcd\"", System.currentTimeMillis() 3000, "func4", null)
};
TriggeredServerCodeResult[] secondResults[] = new TriggeredServerCodeResult[] {
    TriggeredServerCodeResultFactory.create(true, "{\"field\":\"abcd\"}", System.currentTimeMillis() 4000, "func5", null),
    TriggeredServerCodeResultFactory.create(true, "[1, \"2\", 3]", System.currentTimeMillis() 5000, "func6", null),
    TriggeredServerCodeResultFactory.create(false, null, System.currentTimeMillis() 6000, "func7", new ServerError("Error found", "RUNTIME_ERROR", "faital error")),
};
addMockResponseForListTriggeredServerCodeResults(200, firstResults, paginationKey);
addMockResponseForListTriggeredServerCodeResults(200, secondResults, null);

Pair<List<TriggeredServerCodeResult>, String> result1 = api.listTriggeredServerCodeResults(triggerID, 4, null);
Assert.assertEquals(paginationKey, result1.second);
Assert.assertEquals(firstResults.length, result1.first.size());
for (int i = 0 ; i < firstResults.length; ++i) {
    assertSameTriggeredServerCodeResults(firstResults[i], result1.first.get(i));
}

Pair<List<TriggeredServerCodeResult>, String> result2 = api.listTriggeredServerCodeResults(triggerID, 4, result1.second);
Assert.assertNull(result2.second);
Assert.assertEquals(secondResults.length, result2.first.size());
for (int i = 0 ; i < secondResults.length; ++i) {
    assertSameTriggeredServerCodeResults(secondResults[i], result2.first.get(i));
}
```

File: thingif/src/test/java/com/kii/thingif/thingifapi/ListTriggeredServerCodeResultTest.java [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+                TriggeredServerCodeResultFactory.create(true, "\"abcd\"", System.currentTimeMillis() + 3000, "func4", null);
+        TriggeredServerCodeResult serverCodeResult5 =
+                TriggeredServerCodeResultFactory.create(true, "{\"field\":\"abcd\"}", System.currentTimeMillis() + 4000, "func5", null);
+        TriggeredServerCodeResult serverCodeResult6 =
+                TriggeredServerCodeResultFactory.create(true, "[1, \"2\", 3]", System.currentTimeMillis() + 5000, "func6", null);
+        TriggeredServerCodeResult serverCodeResult7 =
+                TriggeredServerCodeResultFactory.create(false, null, System.currentTimeMillis() + 6000, "func7", new ServerError("Error found", "RUNTIME_ERROR", "faital error"));
+
+        this.addMockResponseForListTriggeredServerCodeResults(200, new TriggeredServerCodeResult[]{serverCodeResult1, serverCodeResult2, serverCodeResult3, serverCodeResult4}, paginationKey);
+        this.addMockResponseForListTriggeredServerCodeResults(200, new TriggeredServerCodeResult[]{serverCodeResult5, serverCodeResult6, serverCodeResult7}, null);
```
Comment (shinsuke-nara) [inline thingif/src/test/java/com/kii/thingif/thingifapi/ListTriggeredServerCodeResultTest.java]: Could you remove `this`?

**Comment by shinsuke-nara** [comment]
@yongpingchen 

I found only `bestEffortLimit` test, `bestEffortLimit` and `paginationKey` test and a test without `bestEffortLimit` and `paginationKey` but I do not find only `paginationKey` test. 

Isn't it required?

**Comment by yongpingchen** [comment]
@shinsuke-nara  Fixed. Please confirm. 

>  but I do not find only paginationKey test.

I add test on commit: 8e20e35

**Comment by shinsuke-nara** [comment]
@yongpingchen LGTM!

**Pull request merged**